components/template/Templates.vue

```diff
@@ -37,6 +41,31 @@ import Hero from './Hero.vue'
 export default {
   components: { TemplateCategories, TemplateCard, Hero },
   props: ['activeCategory', 'templates', 'templateCategories'],
+  data() {
+    return {
+      mainTemplateData: this.templates,
+      filteredTemplate: this.templates,
+      searchTerm: '',
+    }
+  },
+  methods: {
+    handleSearch(searchTerm) {
```

Comment:
@hanzalakhan809 I think the search should happen from the backend. In terms of SEO which is good. Confirm this with @dhaladitya108 or @ankitsinghaniyaz